lib/model/IngredientsAnalysisTags.dart

```diff
@@ -1,40 +1,70 @@
-enum VeganStatus { IS_VEGAN, IS_NOT_VEGAN, MAYBE }
+enum VeganStatus {
+  VEGAN,
+  NON_VEGAN,
+  MAYBE_VEGAN,
+  VEGAN_STATUS_UNKNOWN,
+}
 
-enum VegetarianStatus { IS_VEGETARIAN, IS_NOT_VEGETARIAN, MAYBE }
+enum VegetarianStatus {
+  VEGETARIAN,
+  NON_VEGETARIAN,
+  MAYBE_VEGETARIAN,
+  VEGETARIAN_STATUS_UNKNOWN,
+}
 
-enum PalmOilFreeStatus { IS_PALM_OIL_FREE, IS_NOT_PALM_OIL_FREE, MAYBE }
+enum PalmOilFreeStatus {
+  PALM_OIL_FREE,
+  PALM_OIL,
+  MAY_CONTAIN_PALM_OIL,
+  PALM_OIL_CONTENT_UNKNOWN,
+}
 
 class IngredientsAnalysisTags {
   IngredientsAnalysisTags(List<dynamic> data) {
     if (data.contains('en:vegan')) {
-      veganStatus = VeganStatus.IS_VEGAN;
+      veganStatus = VeganStatus.VEGAN;
+    }
+    if (data.contains('en:non-vegan')) {
+      veganStatus = VeganStatus.NON_VEGAN;
+    }
+    if (data.contains('en:maybe-vegan')) {
+      veganStatus = VeganStatus.MAYBE_VEGAN;
+    }
+    if (data.contains('en:vegan-status-unknown')) {
+      veganStatus = VeganStatus.VEGAN_STATUS_UNKNOWN;
     }
 
     if (data.contains('en:vegetarian')) {
-      vegetarianStatus = VegetarianStatus.IS_VEGETARIAN;
+      vegetarianStatus = VegetarianStatus.VEGETARIAN;
+    }
+    if (data.contains('en:non-vegetarian')) {
+      vegetarianStatus = VegetarianStatus.NON_VEGETARIAN;
+    }
+    if (data.contains('en:maybe-vegetarian')) {
+      vegetarianStatus = VegetarianStatus.MAYBE_VEGETARIAN;
+    }
+    if (data.contains('en:vegetarian-status-unknown')) {
+      vegetarianStatus = VegetarianStatus.VEGETARIAN_STATUS_UNKNOWN;
     }
 
     if (data.contains('en:palm-oil-free')) {
-      palmOilFreeStatus = PalmOilFreeStatus.IS_PALM_OIL_FREE;
+      palmOilFreeStatus = PalmOilFreeStatus.PALM_OIL_FREE;
     }
-
-    if (data.contains('en:non-vegan')) {
-      veganStatus = VeganStatus.IS_NOT_VEGAN;
+    if (data.contains('en:palm-oil')) {
+      palmOilFreeStatus = PalmOilFreeStatus.PALM_OIL;
     }
-
-    if (data.contains('en:non-vegetarian')) {
-      vegetarianStatus = VegetarianStatus.IS_NOT_VEGETARIAN;
+    if (data.contains('en:may-contain-palm-oil')) {
+      palmOilFreeStatus = PalmOilFreeStatus.MAY_CONTAIN_PALM_OIL;
     }
-
-    if (data.contains('en:non-palm-oil-free')) {
-      palmOilFreeStatus = PalmOilFreeStatus.IS_NOT_PALM_OIL_FREE;
+    if (data.contains('en:palm-oil-content-unknown')) {
+      palmOilFreeStatus = PalmOilFreeStatus.PALM_OIL_CONTENT_UNKNOWN;
     }
 
-    veganStatus ??= VeganStatus.MAYBE;
+    veganStatus ??= VeganStatus.VEGAN_STATUS_UNKNOWN;
```

Comment:
I would handle the `null` case in `_getVeganStatus`.

Reply:
In fact I kept the code as it was, but we probably should keep null as a possible value, in the case where the API does not send back anything (which should not happen).